File: Array/reverse an array or string.cpp
#include<iostream>
using namespace std;

void reverseArray(int arr[], int n)
{
    int a = n;

    for (int i = 0; i < n; i++)
    {
        if (i>n)
        {
            break;
        }

        int temp = arr[i];
        arr[i] = arr[n];
        arr[n] = temp;
        n--;
    }

    for (int i = 0; i <= a; i++)
    {
        cout<<arr[i];
    }
}

int main()
{
    int n = 5;
    int *arr = new int(n);

    for (int i = 0; i < n; i++)
    {
        cin>>arr[i];
    }

    reverseArray(arr, n-1);

    return 0;
}

File: Array/reverse an array or string.py
## first logic

arr = [1,2,3,4,5]
print("In program")
flag = False
n = len(arr)-1
for i in range(0, len(arr)):
    print(i,n)
    if i>n:
        temp = arr[i]
        arr[i] = arr[n]
        arr[n] = temp
    n-=1
print(arr)